src/Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -1,47 +1,224 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.CommandLine;
 using System.CommandLine.Invocation;
 using System.IO;
+using System.Text;
 using System.Threading.Tasks;
 using Microsoft.DotNet.Interactive;
 using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.CSharp;
 using Microsoft.DotNet.Interactive.Events;
-using Microsoft.DotNet.Interactive.Formatting;
 
 namespace Microsoft.Data.Analysis.Interactive
 {
     public class DataFrameKernelExtension : IKernelExtension
     {
-        public Task OnLoadAsync(IKernel kernel)
+        bool _generateCsvMethod = false;
+
+        public string GetFriendlyName(Type type)
         {
-            //Formatter<DataFrame>.Register((tree, writer) =>
-            //{
-            //    writer.Write("");
-            //}, "text/html");
+            string friendlyName = type.Name;
+            if (type.IsArray)
+            {
+                // Not handled yet
+                return "DataFrameColumn";
+            }
+            if (type.IsGenericType)
+            {
+                int backTick = friendlyName.IndexOf('`');
+                if (backTick > 0)
+                {
+                    friendlyName = friendlyName.Remove(backTick);
+                }
+                friendlyName += "<";
```

Comment:
It feels like there should be an existing utility function somewhere that does this: Given a Type, write out its language specific name.

Note that as written, this will only work for C#.